test/org/mockitousage/annotation/MockWithCustomAnswerAnnotationsTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+
+package org.mockitousage.annotation;
+
+import org.junit.Test;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.exceptions.base.MockitoException;
+import org.mockito.internal.configuration.MockAnnotationProcessor;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.mockitousage.IMethods;
+import org.mockitoutil.TestBase;
+
+import java.lang.reflect.Field;
+import java.util.List;
+
+import static org.mockitousage.annotation.MockAnnotationBuilder.*;
+
+@SuppressWarnings("unchecked")
+public class MockWithCustomAnswerAnnotationsTest extends TestBase {
+
+    public static final int MOCKED_LIST_SIZE = 3;
+    @Mock(customAnswer = MyCustomAnswer.class)
+    IMethods mockWithCustomAnswer;
+
+    @Mock(customAnswer = MyCustomAnswerWithMultipleConstructors.class)
+    IMethods mockWithCustomAnswerHavingMultipleConstructors;
+
+    @Mock(customAnswer = ListCustomAnswer.class)
+    List<String> mockedList;
+
+    @Test
+    public void shouldUseCustomAnswer() {
+        assertSame(MyCustomAnswer.returnedObject, mockWithCustomAnswer.objectReturningMethodNoArgs());
+    }
+
+    @Test
+    public void should_choose_default_constructor() {
+        assertSame(MyCustomAnswer.returnedObject, mockWithCustomAnswerHavingMultipleConstructors.objectReturningMethodNoArgs());
+    }
+
+    @Test
+    public void should_throw_exception_when_custom_answer_cannot_be_instantiated() throws NoSuchFieldException {
+        Field aField = MyCustomAnswer.class.getDeclaredFields()[0];
+
+        try {
+            new MockAnnotationProcessor().process(aMockAnnotation().withCustomAnswer(NotInstantiableClass.class).build(), aField);
```

Comment:
Not sure how that's possible without splitting all the test cases out into their own test classes - consider the following:

````
public class MockWithCustomAnswerAnnotationsTest {
  @Mock(answerClass = NotInstantiableClass.class)
  Object unMockableObject;

  @Mock(answerClass = AnswerThrowingConstructor.class)
  Object objectWithAnswerClassThrowingException;
  ...

  @Test
  public void testNonInstantiableObject() {
    try {
      initMocks(this);
      fail("RuntimeException expected ");
    } catch (MockitoException ex) {
      assertEquals("Custom answer cannot be instantiated",ex.getMessage());
    }
  }

  @Test
  public void testAnswerClassThrowsException() {
    ...
  }
````

Since `initMocks(this);` will initialise all the mocks in the Test Class at once (and, presumably, in unpredictable order), at-most one of the tests will see expected behaviour - the others will see exceptions from instantiating other "failing answer classes". We can't define the @Mock inside the test case, because it's not applicable to a local variable.

Was splitting out test cases to multiple test classes, each with one @Mock, what you intended?